src/Common/src/System/Diagnostics/Debug.Unix.cs

```diff
@@ -11,7 +11,18 @@ namespace System.Diagnostics
     // another version of this partial class with the public visibility 
     static partial class Debug
     {
-        private static string NewLine => "\n"; 
+#if MONO
```

Comment:
I updated my commits, is that more what you were looking for? (https://github.com/dotnet/corefx/pull/17573/files#diff-fe691159eb6eccffbc4791e4ceb1dc66R14)
Because the changes are bigger, I decided to go the other way before, so if you think that the way I did it before is better, then I definitely don't mind reverting.